core/embed/rust/src/ui/model_mercury/component/prompt_screen.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    time::Duration,
+    ui::{
+        component::{Component, Event, EventCtx, PageMsg},
+        display::Color,
+        geometry::{Alignment2D, Offset, Rect},
+        shape,
+        shape::Renderer,
+    },
+};
+
+use super::{theme, Button, ButtonContent, ButtonMsg};
+
+/// Component requesting an action from a user. Most typically embedded as a
+/// content of a Frame and promptin "Tap to confirm" or "Hold to XYZ".
+#[derive(Clone)]
+pub struct PromptScreen {
+    area: Rect,
+    button: Button,
+    circle_color: Color,
+    circle_pad_color: Color,
+    circle_inner_color: Color,
+    dismiss_type: DismissType,
+}
+
+#[derive(Clone)]
+enum DismissType {
+    Tap,
+    Hold,
+}
+
+impl PromptScreen {
+    pub fn new_hold_to_confirm() -> Self {
+        let icon = theme::ICON_SIGN;
+        let button = Button::new(ButtonContent::Icon(icon))
+            .styled(theme::button_default())
+            .with_long_press(Duration::from_secs(2));
+        Self {
+            area: Rect::zero(),
+            circle_color: theme::GREEN,
+            circle_pad_color: theme::GREY_EXTRA_DARK,
+            circle_inner_color: theme::GREEN_LIGHT,
+            dismiss_type: DismissType::Hold,
+            button,
+        }
+    }
+
+    pub fn new_tap_to_confirm() -> Self {
+        let icon = theme::ICON_SIMPLE_CHECKMARK;
+        let button = Button::new(ButtonContent::Icon(icon)).styled(theme::button_default());
+        Self {
+            area: Rect::zero(),
+            circle_color: theme::GREEN,
+            circle_inner_color: theme::GREEN,
+            circle_pad_color: theme::GREY_EXTRA_DARK,
+            dismiss_type: DismissType::Tap,
+            button,
+        }
+    }
+
+    pub fn new_tap_to_cancel() -> Self {
+        let icon = theme::ICON_SIMPLE_CHECKMARK;
+        let button = Button::new(ButtonContent::Icon(icon)).styled(theme::button_default());
+        Self {
+            area: Rect::zero(),
+            circle_color: theme::ORANGE_LIGHT,
+            circle_inner_color: theme::ORANGE_LIGHT,
+            circle_pad_color: theme::GREY_EXTRA_DARK,
+            dismiss_type: DismissType::Tap,
+            button,
+        }
+    }
+}
+
+impl Component for PromptScreen {
+    type Msg = PageMsg<()>;
```

Comment:
I think we can return just `()` until more functionality is needed.

Reply:
https://github.com/trezor/trezor-firmware/pull/3740/commits/807312c855cf325f4457510911a2d8581d03237b